File: Week_02/Seminar/GT/Solutions/FindTheDifference.cpp
class Solution {
public:
    char findTheDifference(string s, string t) {
        int counter[256]{0};
        for(auto& ch: s) {
            counter[ch]++;
        }
        
        for(size_t i = 0; i < t.size(); i++) {
            counter[t[i]]--;
            if(counter[t[i]] == -1) {
                return t[i];
            }
        }
        return -1;
    }
};

File: Week_03/Practicum/PythonAdditional/Find the Difference.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        c1, c2 = [0 for _ in range(26)], [0 for _ in range(26)]

        for c in s:
            c1[ord(c) - ord('a')] += 1
        
        for c in t:
            c2[ord(c) - ord('a')] += 1

        for c in range(26):
            if c1[c] != c2[c]:
                return chr(ord('a') + c)